sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/common/properties/document/impl/DocumentTableColumnsImpl.java

```diff
@@ -72,6 +88,80 @@ private static String[] allColumns(DocumentConfig properties) {
             table.doubleValueColumnName(),
             table.booleanValueColumnName());
     Stream<String> firstConcat = Stream.concat(keyCol, pColumns);
-    return Stream.concat(firstConcat, fixedColumns).toArray(String[]::new);
+    return Stream.concat(firstConcat, fixedColumns).collect(Collectors.toUnmodifiableSet());
+  }
+
+  private static List<Column> allColumns(DocumentConfig config, boolean numberBooleans) {
+    DocumentConfig.DocumentTableConfig table = config.table();
+    int depth = config.maxDepth();
+
+    List<Column> result = new ArrayList<>();
+
+    // key column
+    ImmutableColumn keyColumn =
+        ImmutableColumn.builder()
+            .name(table.keyColumnName())
+            .type(TypeSpecs.format(TypeSpecs.VARCHAR))
+            .kind(Column.Kind.PARTITION_KEY)
+            .build();
+    result.add(keyColumn);
+
+    // path columns
+    List<ImmutableColumn> pathColumns =
+        IntStream.range(0, depth)
+            .mapToObj(i -> table.pathColumnPrefix() + i)
+            .map(
+                pathName ->
+                    ImmutableColumn.builder()
+                        .name(pathName)
+                        .type(TypeSpecs.format(TypeSpecs.VARCHAR))
+                        .kind(Column.Kind.CLUSTERING)
+                        .build())
+            .toList();
+    result.addAll(pathColumns);
```

Comment:
Can't we reuse the name generators here? For example:
```java
    pathColumns(config)
        .forEach(
            pathName ->
                result.add(
                    ImmutableColumn.builder()
                        .name(pathName)
                        .type(TypeSpecs.format(TypeSpecs.VARCHAR))
                        .kind(Column.Kind.CLUSTERING)
                        .build()));
```

Reply:
No because that's the set and here we need order.